lib/datasets/penn-treebank.rb

```diff
@@ -53,19 +55,17 @@ def vocabulary
       end
     end
 
-    def parse_data(data_path, vocab, &block)
+    def parse_data(data_path, vocab)
       index = 0
       File.open(data_path) do |f|
         f.each_line do |line|
           line.split.each do |word|
             word = word.strip
-            x = if vocab.has_key?(word)
-              vocab[word]
-            else
+            unless vocab.key?(word)
               vocab[word] = index
-              index.tap{ index += 1 }
+              index += 1
             end
-            block.call(x)
+            yield(Record.new(word, index))
```

Comment:
`index` will be different from `vocab[word]`.

Reply:
Oh, sorry.